### PR TITLE
Karpenter support with OCI registry, minimum supported version is now v0.17.0 for new cluster creation

### LINKED ISSUE #5764: [Feature] Suport for karpenter v0.17.0
### What feature/behavior/change do you want?
Support v0.17.0 of Karpenter with eksctl.

### Why do you want this feature?
Version 0.17.0 is the latest version of karpenter and i need to use this version for it's recently released features [here](https://github.com/aws/karpenter/releases/tag/v0.17.0).

### PR DESCRIPTION
**Pull request by Himangini**
### Description
Since Karpenter charts are moved to OCI registry we no longer need to do `helm add repo`. Also, versions before `0.17.0` are not available on the new registry. The supported Karpenter version is now changed to the minimum supported version. 
More info here - https://karpenter.sh/docs/upgrade-guide/#upgrading-to-v0170

Closes #5764 

Testing
```
Passing lower versions of Karpenter will now result in error
$ eksctl create cluster -f examples/29-cluster-with-karpenter.yaml 
Error: failed to validate Karpenter config: minimum supported version is v0.17.0

Successful create cluster with Karpenter version 0.17.0
$ eksctl create cluster -f examples/29-cluster-with-karpenter.yaml --verbose 4
2022-11-02 12:09:04 [ℹ]  eksctl version 0.118.0-dev+a7b116d80.2022-11-02T11:58:29Z
2022-11-02 12:09:04 [ℹ]  using region us-west-2
....
2022-11-02 12:31:06 [ℹ]  adding Karpenter to cluster gini-karpenter17-1
2022-11-02 12:31:06 [▶]  cluster endpoint used by Karpenter: https://296467963C8AA8B89F150F38DB1C2EB8.sk1.us-west-2.eks.amazonaws.com
2022-11-02 12:31:06 [▶]  the following chartOptions will be applied to the install: %!w(providers.InstallChartOpts={oci://public.ecr.aws/karpenter/karpenter true karpenter karpenter map[aws:map[defaultInstanceProfile:eksctl-KarpenterNodeInstanceProfile-gini-karpenter17-1] clusterEndpoint:https://296467963C8AA8B89F150F38DB1C2EB8.sk1.us-west-2.eks.amazonaws.com clusterName:gini-karpenter17-1 serviceAccount:map[annotations:map[eks.amazonaws.com/role-arn:arn:aws:iam::083751696308:role/eksctl-gini-karpenter17-1-iamservice-role] create:true name:karpenter]] v0.17.0 0xc0008903c0})
2022-11-02 12:31:10 [▶]  creating 1 resource(s)
2022-11-02 12:31:11 [▶]  creating 1 resource(s)
2022-11-02 12:31:12 [▶]  Clearing discovery cache
2022-11-02 12:31:12 [▶]  beginning wait for 2 resources with timeout of 1m0s
2022-11-02 12:31:38 [▶]  creating 1 resource(s)
2022-11-02 12:31:39 [▶]  creating 15 resource(s)
2022-11-02 12:31:41 [▶]  beginning wait for 15 resources with timeout of 10m0s
2022-11-02 12:31:42 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-02 12:31:44 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-02 12:31:46 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-02 12:31:48 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-02 12:31:50 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-02 12:31:53 [▶]  successfully installed helm chart: karpenter
2022-11-02 12:31:53 [▶]  kubectl: "/usr/local/bin/kubectl"
2022-11-02 12:31:53 [▶]  kubectl version: v1.22.5
2022-11-02 12:31:53 [▶]  found authenticator: aws-iam-authenticator
2022-11-02 12:31:54 [ℹ]  kubectl command should work with "/Users/himangini/.kube/config", try 'kubectl get nodes'
2022-11-02 12:31:54 [✔]  EKS cluster "gini-karpenter17-1" in "us-west-2" region is ready

Successful create cluster with karpenter v0.18.0
$ eksctl create cluster -f examples/29-cluster-with-karpenter.yaml --verbose 4
2022-11-01 10:52:46 [ℹ]  eksctl version 0.118.0-dev+a7b116d80.2022-11-01T10:51:50Z
2022-11-01 10:52:46 [ℹ]  using region us-west-2
...
2022-11-01 11:15:01 [ℹ]  adding Karpenter to cluster gini-karpenter18-1
2022-11-01 11:15:01 [▶]  cluster endpoint used by Karpenter: https://7001095BB3147030C4C994BFAED951B3.gr7.us-west-2.eks.amazonaws.com
2022-11-01 11:15:01 [▶]  the following chartOptions will be applied to the install: %!w(providers.InstallChartOpts={oci://public.ecr.aws/karpenter/karpenter true karpenter karpenter map[aws:map[defaultInstanceProfile:eksctl-KarpenterNodeInstanceProfile-gini-karpenter18-1] clusterEndpoint:https://7001095BB3147030C4C994BFAED951B3.gr7.us-west-2.eks.amazonaws.com clusterName:gini-karpenter18-1 serviceAccount:map[annotations:map[eks.amazonaws.com/role-arn:arn:aws:iam::083751696308:role/eksctl-gini-karpenter18-1-iamservice-role] create:true name:karpenter]] v0.18.0 0xc000b50000})
2022-11-01 11:15:04 [▶]  creating 1 resource(s)
2022-11-01 11:15:06 [▶]  creating 1 resource(s)
2022-11-01 11:15:06 [▶]  Clearing discovery cache
2022-11-01 11:15:06 [▶]  beginning wait for 2 resources with timeout of 1m0s
2022-11-01 11:15:32 [▶]  creating 1 resource(s)
2022-11-01 11:15:33 [▶]  creating 15 resource(s)
2022-11-01 11:15:35 [▶]  beginning wait for 15 resources with timeout of 10m0s
2022-11-01 11:15:35 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-01 11:15:38 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-01 11:15:40 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-01 11:15:42 [▶]  Deployment is not ready: karpenter/karpenter. 0 out of 1 expected pods are ready
2022-11-01 11:15:44 [▶]  successfully installed helm chart: karpenter
2022-11-01 11:15:44 [▶]  kubectl: "/usr/local/bin/kubectl"
2022-11-01 11:15:45 [▶]  kubectl version: v1.22.5
2022-11-01 11:15:45 [▶]  found authenticator: aws-iam-authenticator
2022-11-01 11:15:46 [ℹ]  kubectl command should work with "/Users/himangini/.kube/config", try 'kubectl get nodes'
2022-11-01 11:15:46 [✔]  EKS cluster "gini-karpenter18-1" in "us-west-2" region is ready

```

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: userdocs/src/usage/eksctl-karpenter.md [hunk 1]
```diff
@@ -1,8 +1,13 @@
...
 `eksctl` provides adding [Karpenter](https://karpenter.sh/) to a newly created cluster. It will create all the necessary
-prerequisites outlined in Karpenter's [Getting Started](https://karpenter.sh/docs/getting-started/) section including installing
-Karpenter itself using Helm. We currently support installing versions up to `0.15.0`
+prerequisites outlined in Karpenter's [Getting Started](https://karpenter.sh/docs/getting-started/getting-started-with-eksctl/) section including installing
+Karpenter itself using Helm. We currently support installing versions starting `0.17.0` and above.
+
+!!!note
+With [v0.17.0](https://karpenter.sh/docs/upgrade-guide/#upgrading-to-v0170) Karpenter’s Helm chart package is now stored in Karpenter’s OCI (Open Container Initiative) registry. 
+Eksctl therefore is not supporting lower versions for new cluster creation. Previously created clusters shouldn't be affected by this change. 
+If you wish to upgrade your current installation of Karpenter please refer to the [upgrade guide](https://karpenter.sh/docs/upgrade-guide/)
```
Comment (TiberiuGC) [inline userdocs/src/usage/eksctl-karpenter.md:10]:
```suggestion
!!!note
    With [v0.17.0](https://karpenter.sh/docs/upgrade-guide/#upgrading-to-v0170) Karpenter’s Helm chart package is now stored in Karpenter’s OCI (Open Container Initiative) registry. 
    Eksctl therefore is not supporting lower versions for new cluster creation. Previously created clusters shouldn't be affected by this change. 
    If you wish to upgrade your current installation of Karpenter please refer to the [upgrade guide](https://karpenter.sh/docs/upgrade-guide/)
```

File: pkg/karpenter/karpenter.go [hunk 2]
```diff
@@ -78,15 +76,27 @@ func (k *Installer) Install(ctx context.Context, serviceAccountRoleARN string, i
...
 		CreateNamespace: true,
 		Namespace:       DefaultNamespace,
 		ReleaseName:     releaseName,
 		Values:          values,
 		Version:         k.ClusterConfig.Karpenter.Version,
-	}); err != nil {
+		RegistryClient:  registryClient,
+	}
+
+	logger.Debug("the following chartOptions will be applied to the install: %w", options)
```
Comment (TiberiuGC) [inline pkg/karpenter/karpenter.go:97]:
```suggestion
	logger.Debug("the following chartOptions will be applied to the install: %+v", options)
```

Comment (Himangini) [reply pkg/karpenter/karpenter.go]: done

**Review by TiberiuGC — APPROVED** [review]
LGTM! 👍

**Pull request merged**